### PR TITLE
Change an existing switch statement to the newer syntax

### PR DESCRIPTION
**Pull request by ch4570**
This PR updates the switch statements from Java 11 and earlier, applying modern syntax to eliminate redundant break statements and enhance code readability by making it block-oriented.

### CODE REVIEW

File: spring-core/src/main/java/org/springframework/core/serializer/DefaultDeserializer.java [hunk 1]
```diff
@@ -66,11 +66,9 @@ public DefaultDeserializer(@Nullable ClassLoader classLoader) {
 	 */
 	@Override
 	public Object deserialize(InputStream inputStream) throws IOException {
-		ObjectInputStream objectInputStream = new ConfigurableObjectInputStream(inputStream, this.classLoader);
-		try {
+		try (ObjectInputStream objectInputStream = new ConfigurableObjectInputStream(inputStream, this.classLoader)) {
```
Comment (snicoll) [inline spring-core/src/main/java/org/springframework/core/serializer/DefaultDeserializer.java:69]: This is an unrelated  change from a previous PR. Please revert.

**Review by snicoll — CHANGES_REQUESTED** [review]

**Comment by bclozel** [comment]
This code belongs to our ASM fork, I'm not sure we should apply cosmetic changes there as it's likely to add conflicts when we upgrade.

**Comment by snicoll** [comment]
Ah I had missed that. Yeah, in that case, I am in favor of closing this.
